### ISSUE TITLE
Display the number of tokens in the input / in the conversation so far

### ISSUE THREAD

**Issue by michael-imbeault** [oobabooga/text-generation-webui#2243]
**Description**

Display the number of tokens in the input / in the conversation so far. Since the context window is limited, it is important to know when you exceed it / how many tokens you have left for full context to be considered. This applies to the original input, but also to the full conversation.

**Comment by Ph0rk0z** [comment]
It outputs it to the console.

**Comment by michael-imbeault** [comment]
Yes, so in theory very much possible to have it displayed in the GUI?

**Comment by Ph0rk0z** [comment]
There is a count tokens button in notebook mode. If you are using chat then it won't work.

**Comment by michael-imbeault** [comment]
It is such an important parameter that it should be available on the GUI at all times in my opinion, and updated dynamically if possible.

**Comment by Ph0rk0z** [comment]
I don't disagree and I like how silly tavern counts tokens for character cards and does generation time under the message too. Just nobody has made PRs for that stuff here yet.

**Comment by kha84** [comment]
Indeed, I can see original and new tokens are being calculated (https://github.com/oobabooga/text-generation-webui/blob/main/modules/text_generation.py#L407) - so the question is how can we get them fetched to UI.

In https://github.com/oobabooga/text-generation-webui/blob/main/modules/ui_default.py module, which implements UI for "Default" tab, there's a code that displays count of tokens (see "token-counter" placeholder which gets populated from count_tokens(x) function (https://github.com/oobabooga/text-generation-webui/blob/main/modules/prompts.py#L22)

One needs to copy that "original_tokens" variable (https://github.com/oobabooga/text-generation-webui/blob/main/modules/text_generation.py#L430) to some other global variable so it will be visible from outside the module first. 
Then to introduce the same placeholder to display the count in HTML and function to update it for "Chat" UI (https://github.com/oobabooga/text-generation-webui/blob/main/modules/ui_chat.py)

I guess, that should do the job

**Comment by kha84** [comment]
Raised pull request - https://github.com/oobabooga/text-generation-webui/pull/5357/files
It's kind of hacky way to do that and people are kindly welcomed to fix whatever issues they see with this, but for me it works.
![image](https://github.com/oobabooga/text-generation-webui/assets/110789576/29407218-3f96-4b2c-b9f8-637c71094970)
